### ISSUE TITLE
Add Merge Layer Test Cases

### ISSUE THREAD

**Issue by therealansh** [JetBrains/KotlinDL#276]
KotlinDL is right now does not have Merge Layer test cases like `Add`, `Concatenate`,  `Multiply` and so on. I guess addition of test cases will be great. Also what do you think about refactoring and organising the layer test cases directory @zaleslaw?

**Comment by zaleslaw** [comment]
Yes, a lot of tests are missed for many layers. It will be great to add them to the test suite

**Comment by therealansh** [comment]
Hey @zaleslaw, would it be nice if I put Merge tests in one file as the `ReshapeTest` looks tidy but what are your thoughts?

**Comment by zaleslaw** [comment]
Yes, put it into one file

**Comment by therealansh** [comment]
Hey @zaleslaw , have a look at this test case, this is from tf official [doc](https://www.tensorflow.org/api_docs/python/tf/keras/layers/average) yet it fails, AFAIK the `assertLayerOutputIsCorrect` checks the content of the output as well and `Average()` should produce the same effect as given in the doc. Yet the test fails and when i tried to debug it it just returns the [x1,x2] as result.
```kotlin
@Test
    fun average(){
        val x1 = Array(2){FloatArray(2){1f} }
        val x2 = Array(2){FloatArray(2){0f} }
        val input = arrayOf(x1,x2)
        val expected = Array(2){FloatArray(2){0.5f} }
        assertLayerOutputIsCorrect(Average(), input,expected)
    }
```

**Comment by zaleslaw** [comment]
Thanks for the note! Probably the `Average` layer could be implemented with bug (low probability) or you could find something else (high probability) what influence on the output

**Comment by therealansh** [comment]
Hey @zaleslaw, I guess the bug is in `LayerTest.kt`. Since, Merge layer abstraction has a overloading of `forward` function, i guess we need overload this as well [`getLayerOutputOp()`](https://github.com/JetBrains/KotlinDL/blob/a47add7c20147ab5d43ab1c65010e901582fbf78/api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/LayerTest.kt#L23) as it takes `Array<*>` which returns a Operand<Float> rather than `List<Operand<Float>>` I guess this can be the only reason for it. Otherwise the Average layer is implemented correctly.
